src/Widgets/DataFormContainerWidget/DataFormContainerWidgetComponent.tsx

```diff
@@ -131,9 +141,12 @@ function valueFromElement(
 
   if (element.type === 'file') {
     const files = element.files
-    if (element.multiple) return files ? [...files] : []
-    if (files?.length === 1) return files[0]!
-    return null
+    if (!files) return element.multiple ? [] : null
+
+    const blobToBinaryPromises = [...files].map(blobToBinary)
+    return element.multiple
+      ? Promise.all(blobToBinaryPromises)
+      : blobToBinaryPromises[0]!
```

Comment:
* [ ] The old code did a `length === 1` check to allow for the bang cast. The `files` can be zero-length (https://developer.mozilla.org/en-US/docs/Web/API/HTMLInputElement/files#javascript). We could do `: blobToBinaryPromises[0] || null`.